skimage/segmentation/_chan_vese.py

```diff
@@ -0,0 +1,323 @@
+    """
+    H = heavyside(phi)
+    avgenergy = _cv_difference_from_average_term(img, H, lambda1, lambda2)
+    #lenenergy = _cv_edge_length_term(phi, mu)
+    lenenergy = mu * np.sum(_cv_delta(phi))
+    return np.sum(avgenergy) + lenenergy
+
+
+def _cv_reset_level_set(phi):
+    """
+    This is a placeholder function as resetting the level set is not
+    strictly necessary, and has not been done for this implementation.
+    """
+    return phi
+
+
+def _cv_checkerboard(image_size, square_size):
+    """
+    Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    yv = np.arange(image_size[0]).reshape(image_size[0], 1)
+    xv = np.arange(image_size[1])
+    return (np.sin(np.pi/square_size * yv) *
+            np.sin(np.pi/square_size * xv))
+
+
+def _cv_large_disk(image_size):
+    """
+    Generates a disk level set function. The disk covers the whole
+    image along its smallest dimension.
+    """
+    res = np.ones(image_size)
+    centerY = int((image_size[0]-1) / 2)
+    centerX = int((image_size[1]-1) / 2)
+    res[centerY, centerX] = 0.
+    radius = float(min(centerX, centerY))
+    return (radius-distance(res)) / radius
+
+
+def _cv_small_disk(image_size):
+    """
+    Generates a disk level set function. The disk covers half of
+    the image along its smallest dimension.
+    """
+    res = np.ones(image_size)
+    centerY = int((image_size[0]-1) / 2)
+    centerX = int((image_size[1]-1) / 2)
+    res[centerY, centerX] = 0.
+    radius = float(min(centerX, centerY))
+    return (radius/2 - distance(res)) / (radius*1.5)
+
+
+def _cv_initial_shape(starting_level_set, img):
+    """
+    Generates a checkerboard level set function. According to
+    Pascal Getreuer, such a level set function has fast convergence.
+    """
+    res = []
+    if type(starting_level_set) == str:
+        if starting_level_set == 'checkerboard':
+            res = _cv_checkerboard(img.shape, 5)
+        elif starting_level_set == 'disk':
+            res = _cv_large_disk(img.shape)
+        elif starting_level_set == 'small disk':
+            res = _cv_small_disk(img.shape)
+        else:
+            raise ValueError("Incorrect name for starting level set preset.")
+    else:
+        res = starting_level_set
+    return res
+
+
+def chan_vese(img, mu=0.25, lambda1=1.0, lambda2=1.0, tol=1e-3, max_iter=500,
+              dt=0.5, starting_level_set='checkerboard',
+              extended_output=False):
+    """Segment objects without clearly defined boundaries.
+
+    Active contour model by evolving a level set. Supports 2D
+    grayscale images only, and does not implement the area term
+    described in the original article.
+
+    Parameters
+    ----------
+    img : (M, N) 2D array
+        Grayscale image to be segmented.
+    mu : float
+        'edge length' weight parameter. Higher mu values will produce
```

Comment:
Use `` to highlight the parameters (i.e. """Higher `mu` values..."""). Here and below for `lambda`, etc.